vulcan/lib/client/jsx/components/workflow/user_interactions/inputs/input_types.ts

```diff
@@ -74,15 +75,41 @@ export function bindInputSpecification(input: InputSpecification,
   return {
     ...input,
     onChange(v: Maybe<unknown>) {
-      setInputs(inputs => ({...inputs, [input.source]: v}))
+      if (collapsesOutputs(input.type) && isSome(v)) {
+        const values = Object.entries(withDefault(v, {}) as any).reduce((acc: {[key: string]: unknown}, [key, value]) => {
+          acc[`${input.name}/${key}`] = value;
+
+          return acc;
+        }, {});
+        setInputs(inputs => ({...inputs, ...values as any}))
+      } else {
+        setInputs(inputs => ({...inputs, [input.source]: v}))
+      }
     },
     data: inputData,
-    value: input.source in buffered ? buffered[input.source] :
-      input.source in session.inputs ? some(session.inputs[input.source]) :
+    value: input.source in buffered ?
```

Comment:
So in order to handle multiple outputs in a UI step, I wound up having to also cut into `value`, here (in order to correctly feed the set value back into the UI component), and [sourceNamesOfSteps](https://github.com/mountetna/monoetna/blob/5e2e99d40b21dcb43c7e08972547b5fbabbc1401/vulcan/lib/client/jsx/selectors/workflow_selectors.ts#L295), because it winds up being a single UI widget handling multiple output ... not sure this is the cleanest way to get the functionality, will take a closer look.